### PR TITLE
docs: Further improve Release docs

### PR DESCRIPTION
**Pull request by tofumatt**
Includes proposal of a new versioning scheme and more accurate representation of the release candidate process.

This includes a proposal to move away from versioning what are, in effect, major releases with minor release bumps that reset at `3.9` > `4.0`. In the future we should just do something like what Firefox or React do and increment major versions at every "release".

This would be closer to semantic versioning as we shouldn't be deprecating things with minor versions, but code marked as deprecated in `3.6` gets removed by `3.9`.

Since `4.0` is upcoming anyway I figured now is a great time to make the change. DISCUSS! 😉

### CODE REVIEW

File: docs/reference/release.md [hunk 1]
```diff
@@ -2,127 +2,206 @@
...
+1. writing a release blog post and changelog
+2. bumping the version
+3. tagging the release
+4. building the plugin
+5. publishing the new release to GitHub
+6. committing to the [plugin repository]
+
+### Writing the Release Post and Changelog
+
+1. Open the [list of closed pull requests](https://github.com/WordPress/gutenberg/pulls?utf8=✓&q=is%3Apr+is%3Aclosed+sort%3Acreated-desc+) and filter by the current milestone.
```
Comment (tofumatt) [inline docs/reference/release.md]: This will only work once we get an auto-tagging milestone bot: #9239

File: bin/build-plugin-zip.sh [hunk 2]
```diff
@@ -44,19 +44,12 @@ if [ ! -z "$changed" ]; then
...
-	sleep 2
-fi
-
 # Do a dry run of the repository reset. Prompting the user for a list of all
 # files that will be removed should prevent them from losing important files!
 status "Resetting the repository to pristine condition. ✨"
 git clean -xdf --dry-run
 warning "🚨 About to delete everything above! Is this okay? 🚨"
-echo -n "[Y]es/[N]o: "
+echo -n "[y]es/[N]o: "
```
Comment (gziolo) [inline bin/build-plugin-zip.sh]: It should be lowercase for `N`, too :)

**Review by gziolo — APPROVED** [review]
It all looks good. It's much detailed than what we had before so I would be happy to land it and iterate.

File: bin/build-plugin-zip.sh (hunk 2 above)
Comment (tofumatt) [reply bin/build-plugin-zip.sh]: No is the default, so it's uppercase 😄

**Comment by tofumatt** [comment]
These docs include the notion we should be labelling `4.1` as `5.0`. @mtias: if that's wrong, we should tweak them, but for now I'ma :shipit:

File: docs/reference/deprecated.md [hunk 3]
```diff
@@ -1,4 +1,4 @@
-Gutenberg's deprecation policy is intended to support backwards-compatibility for two minor releases, when possible. The current deprecations are listed below and are grouped by _the version at which they will be removed completely_. If your plugin depends on these behaviors, you must update to the recommended alternative before the noted version.
```
Comment (aduth) [inline docs/reference/deprecated.md]: Was this supposed to be removed without a replacement? It currently reads awkwardly as "intended to support compatibility for releases". I don't think it should have been removed.

**Pull request merged**